### PR TITLE
fixed dhcp server

### PR DESCRIPTION
**Pull request by d78ui98**
- [x] `Vagrantfile` and `install.sh` now loads `NETWORK` from `settings.sh` 
- [x] Network interface connected to internet automatically detected

### CODE REVIEW

**Comment by d78ui98** [comment]
I have hardcoded .234 in the dhcp address like this -
 ```
DHCP_ADDRESS = NETWORK+".234"
```
another option could be to add a line in `settings.sh` for dhcp address and then use it in `Vagrantfile`
Let me know if you any other better idea

**Comment by dashohoxha** [comment]
Why not `NETWORK+".1"`? The DHCP server in this case is also the gateway of the LAN, and usually the IP address of the gateway ends with ".1".

**Comment by dashohoxha** [comment]
You should also configure NAT on the DHCP server, since it is also the gateway.
But maybe on another PR?

**Comment by d78ui98** [comment]
vagrant boxes are by default configured with NAT.

**Comment by dashohoxha** [comment]
> vagrant boxes are by default configured with NAT

I think that we are talking about two different things.
Can you run `iptables-save` inside the box and show the output?

File: dhcp/Vagrantfile [hunk 1]
```diff
@@ -1,22 +1,24 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+load "../settings.sh"
+INTERFACE = %x(  ip route | grep default | cut -d' ' -f5 )
```
Comment (dashohoxha) [inline dhcp/Vagrantfile]: Call `.strip` here, if needed.

**Comment by d78ui98** [comment]
`iptables-save` gives no output.

**Comment by dashohoxha** [comment]
> iptables-save gives no output

Then you don't have NAT installed.
The NAT that you mention is the one that VirtualBox uses to allow your box to access the internet.
The one that I am talking about is the one that allows the client to access internet through your dhcp box.

 - http://www.nairabytes.net/81-linux/418-how-to-set-up-a-nat-router-on-ubuntu-server-16-04
 - https://serverfault.com/questions/564866/how-to-set-up-linux-server-as-a-router-with-nat
 - https://opensourceforu.com/2015/04/how-to-configure-ubuntu-as-a-router/

**Comment by d78ui98** [comment]
Okay. Thanks for the links. I will take a good look and try to implement it.

**Comment by dashohoxha** [comment]
This may also help: https://www.youtube.com/watch?v=4cc-SNuf-B8

**Comment by d78ui98** [comment]
@dashohoxha in the above comment you said configure NAT on dhcp server on another PR.
So should we merge this one?

**Review by dashohoxha — APPROVED** [review]
I am merging it, but don't forget to configure NAT as well.

**Pull request merged**